hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -314,21 +286,67 @@ public Permission getRequiredPermission() {
         return null;
     }
 
-    private class OperationInfo {
-        private final Member member;
-        private final Future future;
+    private final class MultiTargetCallback {
```

Comment:
`MultiTargetCallback` and `SingleTargetCallback` are very similar to ones in `AbstractMultiTargetMessageTask`. Can you extract these classes and reuse?